sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/SettingFieldsExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Azure.Data.AppConfiguration
+{
+    internal static class SettingFieldsExtensions
+    {
+        /// <summary>
+        /// Maps a SettingFields member to its corresponding service name in accordance with the REST API specification.
+        /// </summary>
+        private static readonly IReadOnlyDictionary<SettingFields, string> s_serviceNameMap = new Dictionary<SettingFields, string>()
+        {
+            { SettingFields.Key         , "key"           },
+            { SettingFields.Label       , "label"         },
+            { SettingFields.Value       , "value"         },
+            { SettingFields.ContentType , "content_type"  },
+            { SettingFields.ETag        , "etag"          },
+            { SettingFields.LastModified, "last_modified" },
+            { SettingFields.IsReadOnly  , "locked"        },
+            { SettingFields.Tags        , "tags"          }
+        };
+
+        public static IEnumerable<string> Split(this SettingFields fields)
+        {
+            if (fields == SettingFields.All)
+            {
+                return null;
```

Comment:
It is considered bad form to ever return `null` for an `IEnumerable<T>`.   We should consider changing this to `Enumerable.Empty<string>()` or just `yield break` if we're generating.

Reply:
We still need to pass a null value to the inner generated client. If we decide to return an empty enumerable, calling methods will need to do:
```cs
var splitFields = fields.Split();
splitFields = (splitFields == Enumerable.Empty<string>()) ? null : splitFields;
```
For this reason, I'm already returning the value we need. I can add docstrings to this method to make sure that the atypical return value is properly documented.